### PR TITLE
Dark theme

### LINKED ISSUE #7: dark mode?!
Still in progress/needs to be done:
- [x] Copy button for code blocks
- [x] Posit footer logo
- [x] Callouts

Tested but removed since it isn't part of the theme:
- Quarto tiers (recommend adding a border to the tags so they pop in dark mode - this needs to be opened within the tiers repo)

### LINKED ISSUE #57: Callout colors
**Is your feature request related to a problem? Please describe.**

The default callout colors for warning, cautions, and important are not intuitive and should be based on the level of severity. 

For example:

- Important (don't ignore this info): Yellow or some other color
- Caution (could result in damage to the product/data loss): Orange
- Warning (physical injury/death): Red

**Describe the solution you'd like**

Reconsider color choices

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by AshleyHenry15**
This PR adds a dark theme and resolves other theme-related issues:

Resolves #57 
Resolves #7 

Known issues:
#56 
#59 
Flashing: https://github.com/quarto-dev/quarto-cli/issues/1325

Outstanding items tracking:
- [x] Nav bar drop-down menu: update hover color

Before we merge:
- [x] Update changelog
- [x] Clean up CSS (make sure that I don't have duplicates, that the style sheets follow the same order/org, etc.)
- [x] Maybe clean up examples in files? 

Link to view docs: 
https://ashley.quarto.pub/theme-testing/

### CODE REVIEW

**Review by edavidaja — APPROVED** [review]
more flash than a website in 1999

File: _extensions/posit-docs/theme-dark.scss [hunk 1]
```diff
@@ -0,0 +1,395 @@
...
+$font-family-sans-serif: "Open Sans", "Segoe UI", Roboto, "Helvetica Neue", "Noto Sans", "Liberation Sans", Arial, sans-serif, system-ui;
+$font-family-monospace: "Source Code Pro", monospace;
+$font-family-emoji: "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol", "Noto Color Emoji";
+$font-size-base: 1rem;
+// scss-docs-end font-variables
+
+//font and body color
+
+$body-color: $dark-md-body;
+$popover-bg: $posit-green !default; // check this - citation previews
```
Comment (m--) [inline _extensions/posit-docs/theme-dark.scss:64]:
```
// check this - citation previews
```

does something here still need to be checked?

File: _extensions/posit-docs/theme-dark.scss [hunk 2]
```diff
@@ -0,0 +1,395 @@
...
+    //font-style: inherit;
+}
+
+code span.ot {
+    //color: rgb(60, 222, 222);
+}
+
+code span.st {
+    //color: greenyellow;
+}
```
Comment (m--) [inline _extensions/posit-docs/theme-dark.scss:183]: for the commented-out bits here: do these need to stay (i.e. they are part of something coming soon), or can they be removed?

File: _extensions/posit-docs/theme-dark.scss [hunk 3]
```diff
@@ -0,0 +1,395 @@
...
+
+.code-with-filename strong {
+  color: #6E9AC3;
+}
+
+/* Code block background */
+$code-block-bg: lighten($dark-md-background, 10%);
+
+//$monobackgroundcolor: $dark-md-background !default;
```
Comment (m--) [inline _extensions/posit-docs/theme-dark.scss:207]: here too re: commented out code?

**Review by m-- — APPROVED** [review]
It looks like there are some commented-out lines in the dark mode `.scss` file, not sure if those are still needed or not? Also, there's some inconsistent whitespace formatting in that file, not sure if you want to fix that (either as part of this, or at some point in the future), but i thought i'd mention it. Hooray dark mode! 😎 🎉

File: _extensions/posit-docs/theme-dark.scss (hunk 1 above)
Comment (AshleyHenry15) [reply _extensions/posit-docs/theme-dark.scss:64]: Verified and comment removed! Thank you!

File: _extensions/posit-docs/theme-dark.scss (hunk 2 above)
Comment (AshleyHenry15) [reply _extensions/posit-docs/theme-dark.scss:183]: I think that I had added the styles assuming that they would need updates but I think that I am safe to remove

File: _extensions/posit-docs/theme-dark.scss (hunk 3 above)
Comment (AshleyHenry15) [reply _extensions/posit-docs/theme-dark.scss:207]: I think that I am ok to remove this as well

**Comment by aronatkins** [comment]
Visual feedback for dark-mode. I was testing using Safari. None of this is must-fix; only recording my initial reactions. Also keep in mind that I am not the target audience for this theme.

### Code

The font/color choices for code blocks and inline code make them feel more prominent than the surrounding text. Here's one small example, but it also applies to other occurrences.

image

This felt especially pronounced with the "Output or filenames" code chunk; it drew my eye over everything else on the page.

### Table headers

Table headers feel more visually prominent in dark mode than in light mode. This may be because they use a brighter color than the surrounding text. The fact that they are bold makes them more of a visual draw than the accompanying section header.

image

### Table row highlighting

Table row highlighting feels much more subtle in dark mode than in light mode.

image

### Code-nested tabset

The nested tabset beneath the "code" pill is doing some strange things.

First, there appears to be a tiny underline lingering on the left, perhaps because the bubble around RHEL is too large?

image

Second, on mouseover, the whole set of headings shifts around. This screen recording shows the effect:

tab-hop

### Callout icons

The callout icons probably need to be a lighter color.

image

File: changelog.md [hunk 4]
```diff
@@ -1,3 +1,14 @@
+# unreleased
+
+* Add Posit dark mode theme
+* Resolves Callout colors issue (#57)
+* Fixed several style issues in default theme
+* Fixed several accessibility issues
+
+Known issue:
+
+* Dark mode reloads cause flashing (Quarto has an open issue with plans to address in v1.5 or v16)
```
Comment (aronatkins) [inline changelog.md:10]: nits: Link to the tracking issue, and v1.6, not v16.

File: index-2.qmd [hunk 5]
```diff
@@ -0,0 +1,98 @@
+---
```
Comment (aronatkins) [inline index-2.qmd:1]: nit: Do you intend to link this page from `_quarto.yml` or from any other page? Also consider naming it something other than `index-2.qmd`.

**Review by aronatkins — APPROVED** [review]

**Comment by AshleyHenry15** [comment]
@aronatkins - The nesting of the pills and tabs gets weird.  I am assuming that it is due to the nesting of the different styles. I think that I will remove the example from the file for now until I can dive in and figure out if there is a way to make that work nicely.

- Table header color: I had changed it to white for accessibility but I just tested and it passes with the gray. So, I have removed the white coloring to the `th` class, and it should be fixed now.
- Table rows: Yeah, we went back and forth on the colors for this a lot. I used GitHub tables as an example but the blue complicates things. This will likely be something that I bring up with Veronica for some guidance at some point.

**Pull request merged**